lib/iris/analysis/__init__.py

```diff
@@ -1026,18 +1026,35 @@ def _percentile(data, axis, percent, **kwargs):
     If a new additive dimension is formed, then it will always be the last
     dimension of the resulting percentile data payload.
 
+    Kwargs:
+
+    * percentile_method (string) :
+        Switch to opt between the scipy.stats.mstats.mquantiles method for
+        calculating percentiles, and the much faster np.percentiles method.
+
     """
+    percentile_method = kwargs.get('percentile_method', None)
     # Ensure that the target axis is the last dimension.
     data = np.rollaxis(data, axis, start=data.ndim)
-    quantiles = np.array(percent) / 100.
     shape = data.shape[:-1]
     # Flatten any leading dimensions.
     if shape:
         data = data.reshape([np.prod(shape), data.shape[-1]])
     # Perform the percentile calculation.
-    result = scipy.stats.mstats.mquantiles(data, quantiles, axis=-1, **kwargs)
+    if percentile_method == 'fast':
+        msg = 'Cannot use fast np.percentile method with masked array.'
+        if ma.isMaskedArray(data):
+            raise TypeError(msg)
+        result = np.percentile(data, percent, axis=-1)
```

Comment:
> I think the kwargs should be passed down

:+1:
That makes good sense + from the docs you'd expect it to do that in both cases.

>  how to handle kwargs differently for different implementations

:-1:
I disagree : I don't think that's at all necessary or helpful **unless** your code is specifically using extra args by testing them, adjusting them or automatically generating them.
In my head, if it is clear that `percentile_method` switches between two underlying methods, and that is an explicit user choice, then any extra args are supplied by the same user code, so all the necessary information and responsibility is in the user's hands.  *This* routine knows nothing about those extra args, and doesn't need to, and certainly should not copy documentation relating to the underlying methods (which could change anyway).